.github/workflows/Test-kitctl.yml

```diff
@@ -0,0 +1,48 @@
+name: This workflow automates the build, testing, and bootstrapping of a test environment on Kubernetes
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches-ignore: [main]
+  workflow_dispatch: {}
+jobs:
+   build-cli:
+    strategy:
+      matrix:
+        os: ["darwin", "linux"]
+        arch: ["amd64"]
+    runs-on: ubuntu-latest
+    permissions:
+       id-token: write
+       contents: read
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/cache@v2
+      with:
+        path: |
+          ~/.cache/go-build
+          ~/go/pkg/mod
+          ~/go/bin/
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+    - uses: actions/setup-go@v2
+      with:
+        go-version: '1.17.1'
+    - name: configure aws credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+          role-to-assume: arn:aws:iam::078365735624:role/github-Token-generate
+          role-session-name: samplerolesession
+          aws-region: us-west-2
+    - name: Builds the KIT CLI ,Provisions the Kit Environment and deletes it
+      run: |
+        GOOS=${{ matrix.os }} GOARCH=${{ matrix.arch }} go build -o ./bin/kitctl ./cmd/kitctl/
+        cd /home/runner/work/kubernetes-iteration-toolkit/kubernetes-iteration-toolkit/substrate/cmd/kitctl
+        go build -v .
+        RANDOM_SUFFIX=$(openssl rand -hex 5)
+        ./kitctl bootstrap kit-github-${RANDOM_SUFFIX}
+        export KUBECONFIG=/home/runner/.kit/env/kit-github-${RANDOM_SUFFIX}/etc/kubernetes/admin.conf
+        ./kitctl delete kit-github-${RANDOM_SUFFIX}
```

Comment:
I'd suggest running kubectl command (e.g. kubectl version) here to do a simple sanity check.

Reply:
I was thinking the same thing, I'll make sure to add that step in my next commit.